src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -152,6 +154,87 @@ private MethodBase FunctionLookup(IList args)
             return mi;
         }
 
+        private static Dictionary<Type, int> GetTypeStatisticsForArray(ExprListNode array)
+        {
+            var usageFreq = new Dictionary<Type, int>();
```

Comment:
is the frequency information actually used anywhere ?
Dictionary<Type, int> - the integer

Reply:
As of now, I think it's just the number of keys in the dictionary that is important.

Reply:
Are you suggesting I use a hashset or something instead?